module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,65 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+
+void printMap(const Bitmap& bitmap){
+
+   for(const auto row : bitmap){
+      for(const auto pixel : row){
+            std::cout<<pixel;
+      }
+      std::cout<<'\n';
+   }
+}
+
+CompressedBitmap compressLine(const std::array<uint8_t, width>& row){
+
+   CompressedBitmap result;
+   result.reserve(width);
+   std::pair<uint8_t, uint8_t> buffer{row[0], 0};
+
+   for(const auto pixel : row){
+
+      if(buffer.first == pixel){
+         buffer.second++;
+      }else{
+         result.push_back(buffer);
+         buffer.first = pixel;
+         buffer.second = 1;
+      }
+   }
+
+   result.push_back(buffer);
```

Comment:
it is possible to reconfigure loop to use only one push_back

Reply:
+1